File: abc136/abc136_b.cs
// https://atcoder.jp/contests/abc136/submissions/8119388

using System;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var N = int.Parse(Console.ReadLine());

            var result = 0;
            for (int i = 1; i <= N; i++)
            {
                if (i.ToString().Length % 2 == 1)
                {
                    result++;
                }
            }
            Console.WriteLine(result);
        }
    }
}

File: abc/abc136/abc136_b.fs
// https://atcoder.jp/contests/abc136/submissions/11990727

let N = stdin.ReadLine() |> int

let rec solver s i =
    if i = N + 1 then s
    elif i.ToString().Length % 2 = 1 then solver (s + 1) (i + 1)
    else solver s (i + 1)

solver 0 1 |> stdout.WriteLine
